storage/range_command.go

```diff
@@ -802,7 +802,13 @@ func (r *Range) AdminSplit(args *proto.AdminSplitRequest, reply *proto.AdminSpli
 			return
 		}
 	}
-
+	// First verify this condition so that it will not return
+	// proto.NewRangeKeyMismatchError if splitKey equals to desc.EndKey,
+	// otherwise it will cause infinite retry loop.
```

Comment:
Where does the infinite retry loop occur? Is it due to the creation of a range covering an empty range of keys?